src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -115,6 +119,58 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
                 // This method will return without triggering the build if the submission is not of type TEST.
                 triggerTemplateBuildIfTestCasesChanged(programmingExercise.getId(), result.getId());
             }
+
+            if (!isSolutionParticipation && !isTemplateParticipation) {
+
+                // check if a manual result exists, if so we want to clone and update this with the new updated test-case and sca feedback
+                Optional<ProgrammingExerciseStudentParticipation> studentParticipation = programmingExerciseParticipationService
+                        .findStudentParticipationWithLatestManualResultsAndFeedbacksAndRelatedSubmissionsAndAssessor(participation.getId());
+
+                if (studentParticipation.isPresent() && studentParticipation.get().getResults().stream().findFirst().isPresent()) {
+
+                    Result manualResult = studentParticipation.get().getResults().stream().findFirst().get();
+
+                    // create a new manual result for the same submission as the automatic result
+                    Result newResult = programmingSubmissionService.saveNewEmptyResult(result.getSubmission());
+
+                    newResult.setAssessor(manualResult.getAssessor());
+                    newResult.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+                    // this makes it the most recent result, but optionally keeps the draft state of an unfinished manual result
+                    newResult.setCompletionDate(manualResult.getCompletionDate() != null ? result.getCompletionDate().plusSeconds(1) : null);
```

Comment:
Just a thought: I hope it is safe/tested here that the tutors will not see two unfinished results (and only the latest one is used and submitted).

Reply:
Yes only the latest one is then used for further manual assessments.